acceptance-tests/bosh_helpers.go

```diff
@@ -0,0 +1,353 @@
+	Expect(err).NotTo(HaveOccurred())
+
+	args = append(args, "--vars-store", varsStore.Name())
+	args = append(args, config.BaseManifestPath)
+
+	varsStoreReader := func(target interface{}) error {
+		varsStoreBytes, err := os.ReadFile(varsStore.Name())
+		if err != nil {
+			return err
+		}
+
+		return yaml.Unmarshal(varsStoreBytes, target)
+	}
+
+	return config.boshCmd(boshDeployment, args...), varsStoreReader
+}
+
+type boshInstance struct {
+	AgentID           string `json:"agent_id"`
+	Az                string `json:"az"`
+	Bootstrap         string `json:"bootstrap"`
+	Deployment        string `json:"deployment"`
+	DiskCids          string `json:"disk_cids"`
+	Ignore            string `json:"ignore"`
+	Index             string `json:"index"`
+	Instance          string `json:"instance"`
+	CommaSeparatedIPs string `json:"ips"`
+	ProcessState      string `json:"process_state"`
+	State             string `json:"state"`
+	VMCid             string `json:"vm_cid"`
+	VMType            string `json:"vm_type"`
+}
+
+func (instance boshInstance) ParseIPs() []string {
+	return strings.Split(instance.CommaSeparatedIPs, ",")
+}
+
+func boshInstances(boshDeployment string) []boshInstance {
+	writeLog("Fetching Bosh instances")
+	cmd := config.boshCmd(boshDeployment, "--json", "instances", "--details")
+	session, err := gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(session, time.Minute, time.Second).Should(gexec.Exit(0))
+
+	output := struct {
+		Tables []struct {
+			Rows []boshInstance `json:"Rows"`
+		} `json:"Tables"`
+	}{}
+
+	err = json.Unmarshal(session.Out.Contents(), &output)
+	Expect(err).NotTo(HaveOccurred())
+
+	return output.Tables[0].Rows
+}
+
+func deleteDeployment(boshDeployment string) {
+	By(fmt.Sprintf("Deleting pcap deployment (deployment name: %s)", boshDeployment))
+	cmd := config.boshCmd(boshDeployment, "delete-deployment")
+	session, err := gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
+	Expect(err).NotTo(HaveOccurred())
+	const timeout = 10
+	Eventually(session, timeout*time.Minute, time.Second).Should(gexec.Exit(0))
+}
+
+func login(username, password string) {
+	By(fmt.Sprintf("Calling bosh login (user: %s)", username))
+	cmd := config.boshInteractiveCmd("login")
+
+	stdin, err := cmd.StdinPipe()
+	Expect(err).NotTo(HaveOccurred())
+
+	stdin.Write([]byte(fmt.Sprintf("%s\n", username)))
+	stdin.Write([]byte(fmt.Sprintf("%s\n", password)))
+	stdin.Close()
+
+	session, err := gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
+	Expect(err).NotTo(HaveOccurred())
+	const timeout = 10
+	Eventually(session, timeout*time.Minute, time.Second).Should(gexec.Exit(0))
+}
+
+func writeLog(s string) {
+	ginkgoConfig, _ := GinkgoConfiguration()
+	for _, line := range strings.Split(s, "\n") {
+		fmt.Printf("node %d/%d: %s\n", ginkgoConfig.ParallelProcess, ginkgoConfig.ParallelTotal, line)
+	}
+}
+
+func downloadFile(info pcapInfo, remotePath string, localFile *os.File, permissions os.FileMode) error {
```

Comment:
This function would benefit significantly from godoc, as the term 'download' is logically overloaded and in this case actually means "SCP".

```suggestion
// downloadFile retrieves the file from remotePath via SSH and stores it in localFile with permissions. info contains the remote's credentials and IP address.
// See copyFileFromRemote
func downloadFile(info pcapInfo, remotePath string, localFile *os.File, permissions os.FileMode) error {
```

Reply:
Yes. It is a sibling of [uploadFile](https://github.com/cloudfoundry/haproxy-boshrelease/blob/master/acceptance-tests/https_ext_crt_list_test.go#L420) in HAproxy. There it's even worse, the function is only defined in one test and with no docs at all. I'd say we will have a follow-up discussion on a shared module for these two releases.

Reply:
We have a follow-up for unifying this. But I also don't want to accept new technical dept into pcap because we just copied from HAProxy. What we build here can be the base of the unified version that we'd also use in HAProxy.